data-prepper-pipeline-parser/src/test/java/org/opensearch/dataprepper/pipeline/parser/EnumDeserializerTest.java

```diff
@@ -0,0 +1,166 @@
+
+        Enum<?> result = objectUnderTest.deserialize(jsonParser, deserializationContext);
+
+        assertThat(result, equalTo(enumWithoutJsonCreator));
+    }
+
+    @Test
+    void enum_class_with_invalid_value_and_jsonValue_annotation_throws_IllegalArgumentException() throws IOException {
+        final EnumDeserializer objectUnderTest = createObjectUnderTest(HandleFailedEventsOption.class);
+        final JsonParser jsonParser = mock(JsonParser.class);
+        final DeserializationContext deserializationContext = mock(DeserializationContext.class);
+        when(jsonParser.getCodec()).thenReturn(objectMapper);
+
+        final String invalidValue = UUID.randomUUID().toString();
+        when(objectMapper.readTree(jsonParser)).thenReturn(new TextNode(invalidValue));
+
+        final IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () ->
+                objectUnderTest.deserialize(jsonParser, deserializationContext));
+
+        assertThat(exception, notNullValue());
+        final String expectedErrorMessage = "Invalid value \"" + invalidValue + "\". Valid options include";
+        assertThat(exception.getMessage(), Matchers.startsWith(expectedErrorMessage));
+    }
+
+    @Test
+    void enum_class_with_invalid_value_and_no_jsonValue_annotation_throws_IllegalArgumentException() throws IOException {
+        final EnumDeserializer objectUnderTest = createObjectUnderTest(TestEnum.class);
+        final JsonParser jsonParser = mock(JsonParser.class);
+        final DeserializationContext deserializationContext = mock(DeserializationContext.class);
+        when(jsonParser.getCodec()).thenReturn(objectMapper);
+
+        final String invalidValue = UUID.randomUUID().toString();
+        when(objectMapper.readTree(jsonParser)).thenReturn(new TextNode(invalidValue));
+
+        final IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () ->
+                objectUnderTest.deserialize(jsonParser, deserializationContext));
+
+        assertThat(exception, notNullValue());
+        final String expectedErrorMessage = "Invalid value \"" + invalidValue + "\". Valid options include";
+        assertThat(exception.getMessage(), Matchers.startsWith(expectedErrorMessage));
+    }
+
+    @Test
+    void create_contextual_returns_expected_enum_deserializer() {
+        final DeserializationContext context = mock(DeserializationContext.class);
+        final BeanProperty property = mock(BeanProperty.class);
+
+        final ObjectMapper mapper = new ObjectMapper();
+        final JavaType javaType = mapper.constructType(HandleFailedEventsOption.class);
+        when(property.getType()).thenReturn(javaType);
+
+        final EnumDeserializer objectUnderTest = new EnumDeserializer();
+        JsonDeserializer<?> result = objectUnderTest.createContextual(context, property);
+
+        assertTrue(result instanceof EnumDeserializer);
```

Comment:
```suggestion
        assertThat(result, instanceOf(EnumDeserializer.class));
```

This will output the class type if it fails the assertion.